packages/provider-elasticsearch/src/example-types/metricGroups/pivot.js

```diff
@@ -210,6 +210,55 @@ let createPivotScope = (node, schema, getStats) => {
       }, groups)
     )
 
+  /*
+   *   Get hoisted properties from the groups of a chart,
+   *   passing values also in case this is needed.
+   *
+   *   This is to avoid having issues in which props are not allowed at the same level.
+   *   hoistProps allows groups to hoist items to top of mapping structure and can be
+   *   used for other needs in which hoisting is required.
+   */
+  let getHoistProps = async ({
+    drilldown = { rowDrills: [], columnDrills: [] },
+    groups = { rows: [], columns: [] },
+  }) =>
+    _.reduce(
+      _.merge,
```

Comment:
This is probably simpler with mergeAll - there’s a lot of nesting here that we can reduce by expressing this as a flow

Reply:
I will rewrite this, but the new prettier configs are exploding the code more than it used to also.  We should probably just write a utility function for the drill down marrying of the groups and potential values, as it is needed in a lot of places.

Reply:
I implemented the feedback here in the latest PR